x-pack/elastic-agent/pkg/agent/cmd/diagnostics.go

```diff
@@ -204,6 +204,15 @@ func diagnosticsCollectCmd(streams *cli.IOStreams, fileName, outputFormat string
 		return fmt.Errorf("failed to communicate with Elastic Agent daemon: %w", err)
 	}
 
+	metrics, err := gatherMetrics(innerCtx)
+	if err == context.DeadlineExceeded {
+		return errors.New("timed out after 30 seconds trying to connect to Elastic Agent daemon")
+	} else if err == context.Canceled {
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("failed to communicate with Elastic Agent daemon: %w", err)
```

Comment:
would outputting a partial response would be useful when metrics error? e.g. the metrics can't be harvested, but the config info can be.

Reply:
yeah, that's part of another task: https://github.com/elastic/beats/issues/29818